cio/io_test.go

```diff
@@ -75,25 +82,25 @@ func TestNewAttach(t *testing.T) {
 	attacher := NewAttach(withBytesBuffers)
 
 	fifos, err := NewFIFOSetInDir("", "theid", false)
-	require.NoError(t, err)
+	assert.NilError(t, err)
 
 	io, err := attacher(fifos)
-	require.NoError(t, err)
+	assert.NilError(t, err)
 	defer io.Close()
 
 	producers := setupFIFOProducers(t, io.Config())
 	initProducers(t, producers, expectedStdout, expectedStderr)
 
 	actualStdin, err := ioutil.ReadAll(producers.Stdin)
-	require.NoError(t, err)
+	assert.NilError(t, err)
 
 	io.Wait()
 	io.Cancel()
-	assert.NoError(t, io.Close())
+	assert.NilError(t, io.Close())
 
-	assert.Equal(t, expectedStdout, stdout.String())
-	assert.Equal(t, expectedStderr, stderr.String())
-	assert.Equal(t, expectedStdin, string(actualStdin))
+	assert.Check(t, is.EqualMultiLine(expectedStdout, stdout.String()))
```

Comment:
Can't we detect multiline output? (sorry for being nitty here).

Reply:
No problem, I appreciate the feedback on the new package.

Yes, I think it could be detected. Initially I opted to be more explicit, but I image that if the comparison fails it could check to check if both args are strings, and if they contain at least one `\n` character.

The difference between these functions is just how the output is formatted. `EqualMultiLine` only accepts strings and uses difflib to print a unified diff as the failure message. Maybe a better name would be `EqualWithDiff()`?

Or maybe it would be better to just detect the types in `Equal` and automatically use difflib when appropriate.

Reply:
I tried it out [here](https://github.com/dnephin/gotestyourself/commit/7c069871b35461b7756c0a1f2d78e9c99876c816) and it seems pretty clean. It depends on some other changes, but I think it's fine to remove it.

Reply:
@dnephin That is really awesome.